src/Microsoft.Graph.Core/Extensions/IDecryptableContentExtensions.cs

```diff
@@ -0,0 +1,86 @@
+        /// <returns>Decrypted content as the provided type.</returns>
+        public static async Task<T> Decrypt<T>(this IDecryptableContent encryptedContent, Func<string, string, Task<X509Certificate2>> certificateProvider)
+        {
+            return JsonConvert.DeserializeObject<T>(await encryptedContent.Decrypt(certificateProvider));
+        }
+        /// <summary>
+        /// Validates the signature and decrypted content attached with the notification.
+        /// https://docs.microsoft.com/en-us/graph/webhooks-with-resource-data#decrypting-resource-data-from-change-notifications 
+        /// </summary>
+        /// <param name="certificateProvider">Certificate provider to decrypt the content. The first parameter is the certificate ID provided when creating the subscription. The second is the certificate thumbprint. The certificate WILL be disposed at the end of decryption.</param>
+        /// <returns>Decrypted content as string.</returns>
+        public static async Task<string> Decrypt(this IDecryptableContent encryptedContent, Func<string, string, Task<X509Certificate2>> certificateProvider)
+        {
+            using (var certificate = await certificateProvider(encryptedContent.EncryptionCertificateId, encryptedContent.EncryptionCertificateThumbprint))
+            using (var rsaPrivateKey = RSACertificateExtensions.GetRSAPrivateKey(certificate))
+            {
+                var decryptedSymetrickey = rsaPrivateKey.Decrypt(Convert.FromBase64String(encryptedContent.DataKey), RSAEncryptionPadding.OaepSHA1);
+                using (var hashAlg = new HMACSHA256(decryptedSymetrickey))
+                {
+                    var expectedSignatureValue = Convert.ToBase64String(hashAlg.ComputeHash(Convert.FromBase64String(encryptedContent.Data)));
+                    if (!string.Equals(encryptedContent.DataSignature, expectedSignatureValue))
+                    {
+                        throw new InvalidDataException("Signature does not match");
+                    }
+                    else
+                    {
+                        return Encoding.UTF8.GetString(AESDecrypt(Convert.FromBase64String(encryptedContent.Data), decryptedSymetrickey));
+                    }
+                }
+            }
+        }
+        private static byte[] AESDecrypt(byte[] dataToDecrypt, byte[] key)
+        {
+            try
+            {
+                using (var cryptoServiceProvider = new AesCryptoServiceProvider
+                {
+                    Mode = CipherMode.CBC,
+                    Padding = PaddingMode.PKCS7,
+                    Key = key
+                })
+                {
+                    var numArray = new byte[AESInitializationVectorSize.Value / 8];
```

Comment:
We can get initialization vector size by getting the AES minimum legal block size (128) from the instantiated `cryptoServiceProvider` and dividing it by 8. This removes the need for the lazy initialization at the top of the class.
```suggestion
                    var numArray = new byte[cryptoServiceProvider.LegalBlockSizes[0].MinSize / 8];
```

Reply:
thanks. Actually it doesn't make sense to rely on the min size as the "spec" requires the IV to be 16 bytes